src/test/groovy/graphql/Issue1847.groovy

```diff
@@ -0,0 +1,42 @@
+package graphql
+
+import spock.lang.Specification
+
+class Issue1847 extends Specification {
+    def schema = TestUtil.schema('''
+                type Query {
+                    listWithCircularReference(filter : TypeWithCircularReference) : String 
+                    list(filter : Type) : String 
+                }
+                input TypeWithCircularReference {
+                    circularField : Type
```

Comment:
There's no circular reference here

Reply:
You're right @tbroyer , did some last minute refactoring and lost the circular reference here. Just pushed a new commit that fixes it. Cheers!